### PR TITLE
feat: add backoffFactor parameter

### LINKED ISSUE #5: Add backoffFactor on retry
@davigsousa has suggested on [this tweet](https://twitter.com/davig_sousa/status/1628412957628456961) to implement a backoffFactor parameter on the `retry`.

I think this is a great [inspo](https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html) to implement!
Screenshot 2023-02-22 at 12 29 40

### PR DESCRIPTION
**Pull request by alanneves**
It closes #5 

- [x] - Added backoffFactor parameter
- [x] - Added backoffFactor parameter test (I used the most effective testing method I could think of at the time)
- [x] - Update documentation

### CODE REVIEW

File: src/tests/retry.test.ts [hunk 1]
```diff
@@ -112,4 +112,33 @@ describe('Retry', () => {
     expect(fn).toHaveBeenCalledTimes(2);
     expect(result).toStrictEqual({ success: true });
   })
+
+  it('should back off between retries based on backoffFactor', async () => {
+    const startTime = new Date().getTime();
```
Comment (NicolasLopes7) [inline src/tests/retry.test.ts:117]: I think we should move it closer to the `retryTest` function!

**Review by NicolasLopes7 — CHANGES_REQUESTED** [review]
Looks Great! Thanks for your contribution! 🎉

Just have 1 comment!

File: src/tests/retry.test.ts (hunk 1 above)
Comment (alanneves) [reply src/tests/retry.test.ts]: Oops, true. I fixed it.

**Review by NicolasLopes7 — APPROVED** [review]

**Comment by NicolasLopes7** [comment]
Just realized that the test is flaky [Check it out](https://github.com/NicolasLopes7/ts-async-kit/actions/runs/4245075536/jobs/7380098231)
Screenshot 2023-02-22 at 14 00 56

It's easy to solve! #7

**Pull request merged**